enzyme/test/Enzyme/ForwardMode/calloc.ll

```diff
@@ -0,0 +1,49 @@
+; RUN: %opt < %s %loadEnzyme -enzyme -enzyme-preopt=false -mem2reg -instsimplify -adce -simplifycfg -S | FileCheck %s
+
+
+@enzyme_dupnoneed = dso_local global i32 0, align 4
+
+define dso_local double @f(double %x) {
+entry:
+  %x.addr = alloca double, align 8
+  %data = alloca double*, align 8
+  store double %x, double* %x.addr, align 8
```

Comment:
For simplification, can you remove the extraneous alloas?

Also do you mind making one of the arguments of calloc passed into the function as an argument (to ensure the fix you added here of `getNewFromOriginal` doesn't accidentally pass if removed).

Reply:
done